Automattic/wp-calypso
# DIFM: Adds social profiles step

#### Changes proposed in this Pull Request

* This PR builds on top of #61835.
* This is the UI for "Social Profiles" step for the new DIFM flow. The PR does not implement the complete flow. The new step is accessible when the `signup/redesigned-difm-flow` feature flag is enabled. However, the end-to-end flow will not be testable.
* The "Continue" CTA dispatches an action that saves data to the redux store and submits the step.
* The "Skip" CTA removes data from the redux store and then submits the step.
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/5436027/158113004-8a25071c-ebe6-412a-abbf-43452998988c.png">


#### Testing instructions



* Go to `/start/do-it-for-me/?flags=signup/redesigned-difm-flow`
* Confirm that the UI matches the screenshot shared above.



Related to 0-as-1201933307121196/f
